src/Cardano/BM/Tracer/Class.lhs

```diff
@@ -0,0 +1,59 @@
+\subsection{Cardano.BM.Tracer.Class}
+\label{code:Cardano.BM.Tracer.Class}
+
+\begin{code}
+module Cardano.BM.Tracer.Class
+    ( Tracer (..)
+    , Contravariant(..)
+    , nullTracer
+    , traceWith
+    ) where
+
+import           Data.Functor.Contravariant (Contravariant (..), Op (..))
+
+\end{code}
+
+The notion of a `Tracer` is an action that can be used to observe
+information of interest during evaluation. `Tracer`s can capture (and
+annotate) such observations with additional information from their
+execution context.
+
+\begin{code}
+newtype Tracer m s = Tracer { tracing :: Op (m ()) s }
```

Comment:
We are using `Op` here, which is a slight complexity over the version we discussed last week. Realise that @avieth did this in past; could see that instances of `Op` might make certain transformations (e.g. collecting sequences of traced items into a monoid) - just want to make sure that we clear why the additional complexity.